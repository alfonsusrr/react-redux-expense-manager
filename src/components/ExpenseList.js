import React from "react";
import { connect } from 'react-redux';
import filters from "../reducers/filters";
import getVisibleExpenses from "../selectors/expenses";
import ExpenseListItem from "./ExpenseListItem";

// Regular component
const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {   
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                getVisibleExpenses(props.expenses, props.filters).map((expense) => {
                    return <ExpenseListItem expense={expense} key={expense.id}/>
                })
            )
        }
    </div>
)

// High order component using mapping from state to props
const mapStateToProps = (state) => {
    return {
        expenses: state.expenses,
        filters: state.filters
    }
}

export default connect(mapStateToProps)(ExpenseList)
