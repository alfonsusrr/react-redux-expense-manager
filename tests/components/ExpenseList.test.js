/**
 * @jest-environment jsdom
 */

import React from "react";
import '@testing-library/jest-dom'
import { screen } from '@testing-library/react';
import renderWithProviders from "./utils/render";
import ExpenseList from "../../src/components/ExpenseList";
import { filledStore, emptyStore } from "../fixtures/testStore";

test('Should render ExpenseList with expenses', () => {
    const { asFragment } = renderWithProviders(<ExpenseList/>, { store: filledStore})
    expect(asFragment()).toMatchSnapshot()
})

test('Should render ExpenseList with empty message', () => {
    const { asFragment } = renderWithProviders(<ExpenseList/>, { store: emptyStore})
    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByText(/No expenses/i)).toBeInTheDocument()
})
