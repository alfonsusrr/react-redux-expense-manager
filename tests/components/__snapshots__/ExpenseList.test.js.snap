// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseList with empty message 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Expense List
    </h1>
    <p>
      No expenses
    </p>
  </div>
</DocumentFragment>
`;

exports[`Should render ExpenseList with expenses 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Expense List
    </h1>
    <div>
      <h3>
        <a
          href="/edit/2"
        >
          Household Bill
        </a>
      </h3>
      <p>
        $ 322 - July 21st 2022
      </p>
    </div>
    <div>
      <h3>
        <a
          href="/edit/4"
        >
          Credit Card
        </a>
      </h3>
      <p>
        $ 3123 - July 19th 2022
      </p>
    </div>
    <div>
      <h3>
        <a
          href="/edit/1"
        >
          Rent
        </a>
      </h3>
      <p>
        $ 200 - July 15th 2022
      </p>
    </div>
    <div>
      <h3>
        <a
          href="/edit/3"
        >
          Internet Bill
        </a>
      </h3>
      <p>
        $ 1121 - July 13th 2022
      </p>
    </div>
  </div>
</DocumentFragment>
`;
